### ISSUE TITLE
💅 `noArguments` reports false negatives in nested functions when `arguments` is shadowed

### ISSUE THREAD

**Issue by qraqras** [biomejs/biome#7628]
### Environment information

```bash
Playground
```

### Rule name

noArguments

### Playground link

https://biomejs.dev/playground/?lintRules=noArguments&code=LwAvACAAZgBhAGwAcwBlACAAbgBlAGcAYQB0AGkAdgBlAAoAYwBvAG4AcwB0ACAAbwB1AHQAZQByACAAPQAgAGYAdQBuAGMAdABpAG8AbgAgACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGkAbgBuAGUAcgAgAD0AIABmAHUAbgBjAHQAaQBvAG4AIAAoACkAIAB7AAoAIAAgACAAIAAgACAAIAAgAHIAZQB0AHUAcgBuACAAYQByAGcAdQBtAGUAbgB0AHMAWwAwAF0AOwAgACAALwAvACAAPAAtACAAZQB4AHAAZQBjAHQAIABpAG4AdgBhAGwAaQBkAAoAIAAgACAAIAB9AAoAIAAgACAAIABsAGUAdAAgAGEAcgBnAHUAbQBlAG4AdABzACAAPQAgAFsAIgBhACIALAAgACIAYgAiACwAIAAiAGMAIgBdADsAIAAgAC8ALwAgADwALQAgAHIAZQBkAGUAZgBpAG4AZQAgAGAAYQByAGcAdQBtAGUAbgB0AHMAYAAgAGgAZQByAGUACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAaQBuAG4AZQByACgAYQByAGcAdQBtAGUAbgB0AHMAWwAwAF0AKQA7AAoAfQA7AAoACgAvAC8AIAB0AHIAdQBlACAAcABvAHMAaQB0AGkAdgBlAAoAYwBvAG4AcwB0ACAAbwB1AHQAZQByACAAPQAgAGYAdQBuAGMAdABpAG8AbgAgACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGkAbgBuAGUAcgAgAD0AIABmAHUAbgBjAHQAaQBvAG4AIAAoACkAIAB7AAoAIAAgACAAIAAgACAAIAAgAHIAZQB0AHUAcgBuACAAYQByAGcAdQBtAGUAbgB0AHMAWwAwAF0AOwAKACAAIAAgACAAfQAKACAAIAAgACAAcgBlAHQAdQByAG4AIABpAG4AbgBlAHIAKABhAHIAZwB1AG0AZQBuAHQAcwBbADAAXQApADsACgB9ADsACgA%3D&language=js

### Expected result

Properly determine the scope of `arguments` and report accordingly.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ryumasai** [comment]
After looking into it a bit, this might be an issue with the semantic model.

The `arguments` object is a local variable available within all non-arrow functions. 
Therefore, `arguments` should always resolve within the function.
If it cannot resolve within the function, I believe it should remain unresolved rather than resolving outside the function.
